### DISCUSSION TITLE
Initial impressions

### DISCUSSION THREAD

**Discussion by mgrider** [NobleRobot/NobleEngine#18] [Ideas and Feature Requests]
I've been exploring the template project, and NobleEngine itself a bit, and have some feedback. I'm posting this as a discussion as opposed to opening issues for each of these, because I don't know how useful that would be, or even if you're looking for this kind of feedback. Let me know if you'd like me to create issues for any of these, and I can certainly do that.

1. The example scenes in the template should really have at least one sprite example. They're actually more example menus, and that's fine, but I really there should be an example game, (even if it's maybe just a sprite bouncing around or something, maybe?). 
3. I also think smaller more focused example scenes would be great. It wasn't obvious that each of the example scenes have one menu item that isn't just a transition, and since they do different things, I could imagine a single focused example scene for each.
4. How does NobleEngine work with the SDK menu items? I don’t see any NobleEngine code that relates to those. I could imagine each scene adding/removing its own system menu items if some are set. That would be neat, I think.
5. Speaking of the system menu, I’m seeing a frame rate dip after the system menu closes on the example scenes. Not sure if that’s real or just in the simulator.
6. The Menu in the example scenes doesn’t automatically scroll your menu items the way the PlaydateSDK grid example does. This may not be that important, but would be nice to have, I think. (You can easily reproduce this by changing the font in one of the example scene menus to the system font.)
7. I don’t see anything in the engine about drawing with lines or primitives. Also probably low importance, but I think it would be nice to wrap some of the PlaydateSDK’s drawing to do some of that stuff in-noble-engine.
8. Documentation seems pretty great in most of the engine code, but it is basically non-existent in the `libraries` and `utilities` folders. There are quite a few utilities that are not obvious how you'd use 'em, IMO.

**Comment by Mark-LaCroix** [comment]
1. This is a really good note. `NobleSprite` is a newer feature, and I just haven't updated the example scenes for awhile.
2. I'd love to have a whole suite of example scenes eventually. I agree that organizing them by feature is the right way to do it.
3. Noble Engine doesn't touch the system menu API. It's a pretty good candidate for integration, though, especially doing it per scene as you suggest.
4. Probably worth creating an issue for this, so it goes on my plate.
5. I purposefully did not include scrolling in `Noble.Menu`. The idea is that it's a single use-case implementation of `playdate.gridview`, not a superset of it (unlike `Noble.Input` or `NobleSprite`). I don't remember how strongly I felt about it, though. I can take another look.
6. I hadn't considered this. I think the SDK does a good job on its own for this, and I suppose a separate drawing library might be a better candidate for these features.
7. The third-party libraries I use aren't really documented anywhere officially, and are not on GitHub. They were created and shared in the Playdate dev community before the SDK was public. I see that this is a problem (worth creating an issue), but it probably won't go high on by priority list since it's not my code. Related: Perhaps I should speak with the authors about fully absorbing those libraries into the engine?

**Reply by mgrider** [reply]
Thanks for your reply! Issues #20 and #21 created for points 4 and 7 above. ✅
